src/libbson/src/bson/bson-memory.c

```diff
@@ -32,7 +36,7 @@ BSON_STATIC_ASSERT2 (bson_mem_vtable_t, sizeof (bson_mem_vtable_t) == sizeof (vo
 static void *
 _aligned_alloc_impl (size_t alignment, size_t num_bytes)
 #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L && !defined(_WIN32) && !defined(__ANDROID__) && \
-   !defined(_AIX)
+   !defined(_AIX) && !(defined(__APPLE__) & MAC_OS_X_VERSION_MIN_REQUIRED < 101500)
```

Comment:
```suggestion
   !defined(_AIX) && !(defined(MAC_OS_X_VERSION_MIN_REQUIRED) && MAC_OS_X_VERSION_MIN_REQUIRED < 101500)
```
Use logical OR rather than bitwise OR. Additionally recommend adjusting the condition to test for the definition of `MAC_OS_X_VERSION_MIN_REQUIRED` instead of `defined(__APPLE__)` (already used to condition the inclusion of `<AvailabilityMacros.h>` above) to guard against the possibility of `OSX_DEPLOYMENT_TARGET` and/or `-mmacosx-version-min` interfering with the presence of the macro (not a configuration space we control nor test).

Reply:
Noting here for future reference that the value of `101500` (MacOS 10.15) is consistent with [libc++'s own checks](https://github.com/llvm/llvm-project/blob/db60244519023a2b083caa3ed3a27a6b59eb03d8/libcxx/include/__config#L662) for `aligned_alloc` support:
```cpp
// It is not yet possible to use aligned_alloc() on all Apple platforms since
// 10.15 was the first version to ship an implementation of aligned_alloc().
#  if defined(__APPLE__)
#    if (defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) &&                                                     \
         __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 101500) ||                                                    \
        /* ... */
#      define _LIBCPP_HAS_C11_ALIGNED_ALLOC 0
#    else
#      define _LIBCPP_HAS_C11_ALIGNED_ALLOC 1
#    endif
#  elif /* ... */
```